flang/lib/Lower/OpenMP.cpp

```diff
@@ -2475,39 +2481,43 @@ static void genBodyOfTargetDataOp(
     genNestedEvaluations(converter, eval);
 }
 
+struct GenOpWithBodyInfo {
```

Comment:
This structure contains the same fields as `CreateBodyOfOpInfo` except for the `args` and `dsp` fields. Maybe it makes sense to have that structure inherit from this one so we're not using different names/types for the same things (e.g. `Location &loc` vs `Location currentLocation`).